hail/src/main/scala/is/hail/expr/ir/streams/EmitStream.scala

```diff
@@ -1294,19 +1292,29 @@ object EmitStream {
             val lEltType = leftProducer.element.emitType
             val rEltType = rightProducer.element.emitType
 
-            // these variables are used as inputs to the joinF
-
             def compare(cb: EmitCodeBuilder, lelt: EmitValue, relt: EmitValue): Code[Int] = {
               assert(lelt.emitType == lEltType)
               assert(relt.emitType == rEltType)
-
-              val lhs = lelt.map(cb)(_.asBaseStruct.subset(lKey: _*))
-              val rhs = relt.map(cb)(_.asBaseStruct.subset(rKey: _*))
-              StructOrdering.make(lhs.st.asInstanceOf[SBaseStruct], rhs.st.asInstanceOf[SBaseStruct],
-                cb.emb.ecb, missingFieldsEqual = false)
-                .compare(cb, lhs, rhs, missingEqual = false)
+              if (x.isIntervalJoin) {
+                val lhs = cb.memoize(lelt.toI(cb).flatMap(cb)(_.asBaseStruct.loadField(cb, lKey(0))))
+                val rhs = cb.memoize(relt.toI(cb).flatMap(cb)(_.asBaseStruct.loadField(cb, rKey(0))))
+                val result = cb.newLocal[Int]("SJRD-interval-compare-result")
+                rhs.toI(cb).consume(cb, {
+                  cb.assign(result, -1)
```

Comment:
Should the interval comparison functions just support taking in missing values? That way the result of comparison against missing is consistent if used in other places in code? Not sure if it should always be handled the same way, this is just me asking.

Reply:
I also find this unsatisfying. The problem is that the interval itself could be missing, and the endpoints could be missing, and it's not obvious to me that those layers of missingness will always be handled the same way. Right now, the interval comparison functions handle the inner missingness.

In fact, I'm now not confident that `missingEqual = false` is right for the inner comparisons. We use that in joins so that we don't join missing keys on the left with missing keys on the right. But should we consider a key `NA(Int32)` on the left as being in the interval `[5, NA]` on the right? I'll go look for what we do currently.